lib/ansible/modules/cloud/openstack/os_port.py

```diff
@@ -356,8 +356,7 @@ def main():
 
         # Neutron API accept 'binding:vnic_type' as an argument
         # for the port type.
-        module.params['binding:vnic_type'] = module.params['vnic_type']
-        module.params.pop('vnic_type', None)
+        module.params['binding:vnic_type'] = module.params.pop('vnic_type')
```

Comment:
default is missing to allow skipping this param

Reply:
> default is missing to allow skipping this param

by intention, because the key should exist. From my point of view, it is a good style to explode, if the key is expected, but missing. But I am open to keeping the default value if you think this would be good here.

Reply:
Well, if you think it is mandatory - you need to tell this to ansible (line 329). However Neutron explicitly states it is optional (see https://developer.openstack.org/api-ref/network/v2/index.html?expanded=create-port-detail#create-port). Parameter recently added can not be mandatory not breaking older clients.

Reply:
Line 329 defines 'None' as the default value, so the key will be in dict, even the value might be none.